hydromt/data_catalog.py

```diff
@@ -1342,53 +1347,67 @@ def get_rasterdataset(
-            if isinstance(data_like, str) and data_like in self.sources:
-                name = data_like
-                source = self.get_source(name, provider=provider, version=version)
+            if isinstance(data_like, (str, Path)):
+                if isinstance(data_like, str) and data_like in self.sources:
+                    name = data_like
+                    source = self.get_source(name, provider=provider, version=version)
+                else:
+                    if "provider" not in kwargs:
+                        kwargs.update({"provider": "user"})
+                    source = RasterDatasetAdapter(path=str(data_like), **kwargs)
+                    name = basename(data_like)
+                    self.add_source(name, source)
+            elif isinstance(data_like, (xr.DataArray, xr.Dataset)):
+                data_like = RasterDatasetAdapter._slice_data(
+                    data_like,
+                    variables,
+                    geom,
+                    bbox,
+                    buffer,
+                    align,
+                    time_tuple,
+                    logger=self.logger,
+                )
+                if has_no_data(data_like):
+                    raise NoDataException()
+                ds = RasterDatasetAdapter._single_var_as_array(
+                    data_like, single_var_as_array, variables
+                )
+                if has_no_data(data_like):
+                    raise NoDataException()
+                return ds
             else:
-                if "provider" not in kwargs:
-                    kwargs.update({"provider": "user"})
-                source = RasterDatasetAdapter(path=str(data_like), **kwargs)
-                name = basename(data_like)
-                self.add_source(name, source)
-        elif isinstance(data_like, (xr.DataArray, xr.Dataset)):
-            data_like = RasterDatasetAdapter._slice_data(
-                data_like,
-                variables,
-                geom,
-                bbox,
-                buffer,
-                align,
-                time_tuple,
-                handle_nodata,
+                raise ValueError(
+                    f'Unknown raster data type "{type(data_like).__name__}"'
+                )
+
+            obj = source.get_data(
+                bbox=bbox,
+                geom=geom,
+                buffer=buffer,
+                zoom_level=zoom_level,
+                align=align,
+                variables=variables,
+                time_tuple=time_tuple,
+                single_var_as_array=single_var_as_array,
+                cache_root=self._cache_dir if self.cache else None,
+                handle_nodata=NoDataStrategy.RAISE,
                 logger=self.logger,
             )
-            return RasterDatasetAdapter._single_var_as_array(
-                data_like, single_var_as_array, variables
+            return obj
+        except NoDataException:
```

Comment:
Is it needed to handle nodata both in get_rasterdataset and rasterdataset.get_data? Or are we doing things twice? Ie you should pass handle_nodata in get_rasterdataset but the result of rasterdataset.get_data should already be either data, None or Error depending on the handle_nodata argument passed?
Same remark for all get_* methods in this file.